src/sidebar.cls.php

```diff
@@ -0,0 +1,174 @@
+<?php
+/**
+ * The Sidebar class
+ *
+ * @since
+ * @package     LiteSpeed
+ * @subpackage  LiteSpeed/inc
+ * @author      LiteSpeed Technologies <info@litespeedtech.com>
+ */
+namespace LiteSpeed;
+
+defined( 'WPINC' ) || exit;
+
+class Sidebar extends Base {
+
+  /**
+   * Add a post meta box into the sidebar of post editor.
+   *
+   * @since    4.3
+   */
+  function add_meta_boxes() {
+    Debug2::debug( '[Sidebar] add_meta_boxes ');
+    add_meta_box(
+      'litespeed_metabox',                      // Unique ID
+      'Litespeed Cache',                        // Title
+      array( Sidebar::cls(), 'meta_box_html' ), // Callback function
+      null,                                     // Admin page (or post type)
+      'side',                                   // Context
+      'high'                                    // Priority
+    );
+  }
+
+  /**
+   * Display the post meta box.
+   *
+   * @since    4.3
+   */
+  function meta_box_html( $post ) {
+    $post_meta = get_post_meta( $post->ID, 'litespeed_metabox', true );
+    $desktop_organized_post_meta = '';
+    $mobile_organized_post_meta = '';
+    $desktop_warning_msg = 'Desktop VPI is empty';
+    $mobile_warning_msg = 'Mobile VPI is empty';
+    if ( isset( $post_meta['desktop'] ) && is_array( $post_meta['desktop'] ) ) {
+      $desktop_organized_post_meta = esc_attr( implode("\n", $post_meta['desktop'] ) );
+      $desktop_warning_msg = '';
+    }
+    if ( isset( $post_meta['mobile'] ) && is_array( $post_meta['mobile'] ) ) {
+      $mobile_organized_post_meta = esc_attr( implode("\n", $post_meta['mobile'] ) );
+      $mobile_warning_msg = '';
+    }
+    wp_nonce_field( basename( __FILE__ ), 'litespeed_metabox_nonce' );?>
+
+    <!-- html starts here -->
+    <h4 for="litespeed_metabox">VPI</h4>
```

Comment:
1. For HTML content, separate it from PHP classes. You can put the following to a new folder like `/tpl/editor`.
2. You were right. We need to only show this VPI settings when VPI is ON.

Reply:
under tml I have a folder called editor that has a file editor_sidebar.php
it has `<?php
namespace LiteSpeed;
defined( 'WPINC' ) || exit;
?>

    <h4 for="litespeed_metabox">VPI</h4>
    <p>
      <label for="vpi_desktop">Desktop Viewport Images</label>
      <?php if ( isset( $post_meta['desktop_timestamp'] ) ) : ?>
        <label id="vpi_desktop_timestamp" style="color:orange;font-size:8px;">Timestamp <?php echo date( 'm/d/Y', $post_meta['desktop_timestamp'] ); ?></label>
      <?php endif; ?>
      <div id="vpi_desktop_edit_box">
        <textarea id="vpi_desktop" name="vpi_desktop" rows="3" cols="30"><?php echo $desktop_organized_post_meta; ?></textarea>
        <p style="color:red;text-align:center;font-size:10px;"><?php echo $desktop_warning_msg; ?></p>
      </div>
    </p>
    <p>
      <label for="vpi_mobile">Mobile Viewport Images</label>
      <?php if ( isset( $post_meta['mobile_timestamp'] ) ) : ?>
        <label id="vpi_mobile_timestamp" style="color:orange;font-size:8px;">Timestamp <?php echo date( 'm/d/Y', $post_meta['mobile_timestamp'] ); ?></label>
      <?php endif; ?>
      <div id="vpi_mobile_edit_box">
        <textarea id="vpi_mobile" name="vpi_mobile" rows="3" cols="30"><?php echo $mobile_organized_post_meta; ?></textarea>
        <p style="color:red;text-align:center;font-size:10px;"><?php echo $mobile_warning_msg; ?></p>
      </div>
    </p>`

Reply:
I don't know how to call this file in Editor class

Reply:
https://github.com/litespeedtech/lscache_wp/blob/master/src/admin-display.cls.php#L529 check how other templates are used.

Reply:
i see thanks